Tests/Marketplace/upload_packs.py

```diff
@@ -464,14 +465,51 @@ def update_index_with_priced_packs(private_storage_bucket: Any, extract_destinat
         logging.info("get_private_packs")
         private_packs = get_private_packs(private_index_path, pack_names,
                                           extract_destination_path)
+
+        logging.info("get_updated_private_packs")
+        updated_private_packs = get_updated_private_packs(private_packs, index_folder_path)
+
         logging.info("add_private_packs_to_index")
         add_private_packs_to_index(index_folder_path, private_index_path)
         logging.info("Finished updating index with priced packs")
     except Exception:
         logging.exception('Could not add private packs to the index.')
     finally:
         shutil.rmtree(os.path.dirname(private_index_path), ignore_errors=True)
-        return private_packs, private_index_path, private_index_blob
+        return private_packs, private_index_path, private_index_blob, updated_private_packs
+
+
+def get_updated_private_packs(private_packs, index_folder_path):
+    """ Checks for updated private packs by compering contentCommitHash between public index json and private pack
+    metadata files.
+
+    Args:
+        private_packs (list): List of dicts containing pack metadata information.
+        index_folder_path (str): The public index folder path.
+
+    Returns:
+        updated_private_packs (list) : a list of all private packs id's that were updated.
+
+    """
+    updated_private_packs = []
+
+    with open(os.path.join(index_folder_path, f"{GCPConfig.INDEX_NAME}.json")) as public_index_file:
+        public_index_json = json.load(public_index_file)
+    public_packs = public_index_json.get("packs", {})
```

Comment:
```suggestion
    private_packs_from_public_index = public_index_json.get("packs", {})
```
naming here is a bit confusing, let's replace it with something more informative, here's a suggestion

Reply:
Done